rsocket/RSocket.cpp

```diff
@@ -1,12 +1,55 @@
 // Copyright 2004-present Facebook. All Rights Reserved.
 
+#include <folly/io/async/EventBaseManager.h>
+
 #include "rsocket/RSocket.h"
 
 namespace rsocket {
 
-std::unique_ptr<RSocketClient> RSocket::createClient(
-    std::unique_ptr<ConnectionFactory> connectionFactory) {
-  return std::make_unique<RSocketClient>(std::move(connectionFactory));
+folly::Future<std::shared_ptr<RSocketClient>> RSocket::createConnectedClient(
+    std::unique_ptr<ConnectionFactory> connectionFactory,
+    SetupParameters setupParameters,
+    std::shared_ptr<RSocketResponder> responder,
+    std::unique_ptr<KeepaliveTimer> keepaliveTimer,
+    std::shared_ptr<RSocketStats> stats,
+    std::shared_ptr<RSocketNetworkStats> networkStats,
+    std::shared_ptr<ResumeManager> resumeManager,
+    std::shared_ptr<ColdResumeHandler> coldResumeHandler,
+    OnRSocketResume) {
+  auto c = std::shared_ptr<RSocketClient>(new RSocketClient(
```

Comment:
that's true, consider adding a create() method to RSocketClient that creates it. `make_shared()` saves an allocation.